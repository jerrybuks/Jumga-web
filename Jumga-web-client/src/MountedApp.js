import React, { Suspense } from 'react';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { compose } from "redux";
import { selectCurrentUser, selectIsFetchingUser } from './redux/user/user.selectors';

import Spinner from './components/spinner/Spinner';
import WithSpinner from './components/with-spinner/withSpinner';

const AdminAuthenticatedApp = React.lazy(() => import('./routing/AdminAuthenticatedApp'))
const UserAuthenticatedApp = React.lazy(() => import('./routing/UserAuthenticatedApp'))
const UnAuthenticatedApp = React.lazy(() => import('./routing/UnAuthenticatedApp'))


export function MountedApp({ currentUser }) {

	return (
		<>
            <Suspense fallback={<Spinner />}>
                {currentUser ? currentUser.isAdmin ? (
                    <AdminAuthenticatedApp />
                ) : (
                    <UserAuthenticatedApp user={currentUser} />
                ) : (
                    <UnAuthenticatedApp />
                )}
            </Suspense>
		</>
	);
}

const mapStateToProps = createStructuredSelector({
	currentUser: selectCurrentUser,
	isLoading: selectIsFetchingUser
});

const MountedAppConatianer = compose(
	connect(mapStateToProps),
	WithSpinner
  )(MountedApp);

export default MountedAppConatianer;
