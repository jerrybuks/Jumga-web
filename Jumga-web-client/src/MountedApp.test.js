import React from 'react';
import {MountedApp} from './MountedApp';
import { shallow } from 'enzyme';


describe('MountedApp component', () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      currentUser : {}
    };
    wrapper = shallow(<MountedApp {...mockProps} />);
  });

  it('should render MountedApp component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
