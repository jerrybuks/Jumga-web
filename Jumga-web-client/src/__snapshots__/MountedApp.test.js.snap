// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MountedApp component should render MountedApp component 1`] = `ShallowWrapper {}`;
